Mathlib/Topology/Basic.lean

```diff
@@ -570,6 +570,14 @@ theorem dense_compl_singleton_iff_not_open :
     obtain rfl : U = {x} := eq_singleton_iff_nonempty_unique_mem.2 ⟨hne, hUx⟩
     exact ho hU
 
+/-- If a closed property holds for a dense subset, it holds for the whole space. -/
+@[elab_as_elim]
+lemma Dense.induction (hs : Dense s) {P : X → Prop}
```

Comment:
As this can be used as
```
  induction x using hs.induction with
  | H x hx => sorry
  | isClosed => sorry
```
it maybe makes sense to rename `H` to `mem`